docs/rfcs/rfc-011-performance-testing.md

```diff
@@ -0,0 +1,67 @@
+
+### Resolvers execution
+By running simple JavaScript performance tests over our resolvers with complex schemas we should be able to find bottlenecks in our code that may improve overall performance of a running server.
+
+> Note: Despite these improvements directly affecting user servers, it's impact is probably very low compared to Cypher improvements. Because of this, these tests should probably be of lower priority.
+### E2E performance
+A full E2E test, in which time spanned for a real-life case covering all our stack (JavaScript, GraphQL and Neo4j) should be considered as a long-term goal. This kind of test would be one of the best indicators of overall performance, but requires a strict control of all the stack and will be the harder to implement.
+
+## Augmented schema
+There are some performance measurements related to the augmented schema generation process that we should consider. 
+
+### Startup time
+The generation of the augmented schema is a one-off execution at server startup, but due to the long execution of this process for large schemas, it may be worth optimising this process to speed up server startup, even if it won't produce a real difference in a running server performance.
+
+This kind of performance is a pure time span performance of JavaScript execution, the same principles as Server performance apply.
+
+### Schema size
+Another measurable metric is the total size of the augmented schema. This schema lives in memory through the server's life, optimising the size may improve performance, particularly on low-memory servers.
+
+The resulting size of the augmented schema can be easily measured by generating a big schema (e.g. [FHIR Schema](https://gist.github.com/aspectgfg/087612013e61eac4f7dffff1dc372a5e)) an simply counting the size of the resulting file (after minification).
+
+> Note: Because the schema lives in memory under the GraphQL library internal representation, the metric of the raw schema may not be an accurate measure of the in-memory gains.
+## Challenges
+In addition to the challenges present in every testing strategy, we may find the following challenges when building any performance testing suite:
+* **Continuous Integration**: We should be able to measure and validate performance improvements, as well as detect noticeable drops in performance when developing new features.
+* **Benchmarks**: We need to find reasonable benchmarks and goals to hit in terms of performance (i.e. What is the performance we want to achieve for x feature?).
+* **Standard measures**: Any time-based measure is bound to hardware dependencies, for CI and long-term purposes, these measures should happen in a strictly controlled machine and standard benchmarking practices (e.g. average multiple runs) should be taken into consideration.
+
+## Risks and unknowns
+* Currently, `@neo4j/graphql` works on the principle of a single Cypher query per request, we need to validate if this is still the best way to achieve good performance.
```

Comment:
**question**: Are there any security considerations we have to discuss or keep an eye out in the future?

Reply:
Good point, I don't think there are any for the scope of this PR, as we only intend to benchmark, not change the queries.

Security should be discussed on each performance improvement as needed IMO